From 0174d1a8356374ac45bb1c8c148d1315d685e7c4 Mon Sep 17 00:00:00 2001
From: CrystalPT <CrystalP@xbmc.org>
Date: Sun, 2 Jan 2011 03:06:57 -0500
Subject: [PATCH] fixed: bitstream mode improperly set. Ticket #10981.

Broke AC3 passthrough for emotiva and NAD receivers.
Patch sent to ffmpeg by Anssi and accepted.
---
 libavformat/spdif.c    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libavformat/spdif.c b/libavformat/spdif.c
index 98184d6..0dd543e 100644
--- a/libavformat/spdif.c
+++ b/libavformat/spdif.c
@@ -108,7 +108,7 @@ static void bswap_buf16(uint16_t *dst, const uint16_t *src, int w)
 static int spdif_header_ac3(AVFormatContext *s, AVPacket *pkt)
 {
     IEC958Context *ctx = s->priv_data;
-    int bitstream_mode = pkt->data[6] & 0x7;
+    int bitstream_mode = pkt->data[5] & 0x7;
 
     ctx->data_type  = IEC958_AC3 | (bitstream_mode << 8);
     ctx->pkt_offset = AC3_FRAME_SIZE << 2;
-- 
1.7.2.3.msysgit.0

